Automattic/jetpack
# WPCOM Marketplace: Add updater for providing updates for Marketplace Products

Fixes https://github.com/Automattic/dotcom-forge/issues/3228

## Proposed changes:

* This PR adds the functionality to provide automated updates to WPCOM Marketplace plugins/themes. This PR hooks into the core's update routine to inject updates for plugins/themes. 

## Testing instructions:






* Use WoA Dev blog for testing. 
* Purchase WooCommerce Bookings plugin from WPCOM Marketplace.
* SSH into your dev blog.
* Manually update version of WooCommerce Bookings plugin to mark it outdated. ie change version to `2.0.1` from `2.0.2`. 
```
vim ~/htdocs/wp-content/plugins/woocommerce-bookings/woocommerce-bookings.php
```
* Visit plugins page of test site. You should be able to see update related to WooCommerce Bookings. Don't update the plugin yet.
* SSH into your dev blog.
* Run `wp shell` and then run following command. The output would look something like one below.
```
get_site_transient( 'update_plugins', 'site-transient' )->response
```
<img width="659" alt="Screenshot 2023-09-06 at 3 17 10 PM" src="https://github.com/Automattic/jetpack/assets/86406124/f02108b3-068d-4340-8f9a-2a5efe11f951">

* This update is provided by WooCommerce plugin for Woo plugins. We want to overide this update if exists else push new update for WPCOM marketplace products. 
* Install, and activate [the Jetpack Beta plugin](https://jetpack.com/download-jetpack-beta/). Once the plugin is active, go to Jetpack > Jetpack Beta > WordPress.com Features and enable the `add/marketplace-products-updater` branch.
* Add following constant in `wp-config.php` file. This would allow you to test feature branches using jetpack beta. 
```
define( 'JETPACK_MU_WPCOM_LOAD_VIA_BETA_PLUGIN', 1 );
```
* Run `wp shell` and then run following command. This would force updates check to run again. 
```
delete_site_transient( 'update_plugins' )
```
* Go to plugins on the site and reload the page. 
* Close all existing session of `wp shell`, start new session `wp shell` and then run following command. The output would look something like one below.
```
get_site_transient( 'update_plugins', 'site-transient' )->response
```
<img width="732" alt="Screenshot 2023-09-06 at 3 30 39 PM" src="https://github.com/Automattic/jetpack/assets/86406124/6cf8a09d-aef8-436c-a09d-f3a3c837671a">

* As you can see in above image that `Upgrade_notice` and `id` does not exists on the update object. This means that the updates was provided by this PR.
